File: customview/src/main/java/dinson/customview/http/manager/LoggingInterceptor.java
package dinson.customview.http.manager;

import java.io.IOException;
import java.util.Locale;

import dinson.customview.utils.LogUtils;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * 拦截请求日志
 *
 * @author Dinson - 2017/7/25
 */
public class LoggingInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        LogUtils.i("**********************************************************************");
        LogUtils.d(request.toString());
        String headers = formetStringArray(request.headers().toString().split("\n"));
        if (!headers.equals("[]")) LogUtils.d(headers);

        long t1 = System.nanoTime();
        okhttp3.Response response = chain.proceed(chain.request());
        long t2 = System.nanoTime();

        LogUtils.d(String.format(Locale.getDefault(), "Received response for in %.1fms", (t2 - t1) / 1e6d));
        LogUtils.d(formetStringArray(response.headers().toString().split("\n")));

        okhttp3.MediaType mediaType = response.body().contentType();
        String content = response.body().string();
        LogUtils.d("response body:" + content);
        LogUtils.i("**********************************************************************");
        return response.newBuilder()
            .body(okhttp3.ResponseBody.create(mediaType, content))
            .build();
    }


    private String formetStringArray(String[] arr) {
        StringBuffer stringBuffer = new StringBuffer();
        for (String s : arr) {
            String[] split = s.split(":", 2);
            String date = split.length == 2 ? split[1].trim() : split[0].trim();
            stringBuffer.append("[" + date + "]");
        }
        return stringBuffer.toString();
    }

}

File: customview/src/main/java/dinson/customview/http/manager/LoggingInterceptor.kt
package dinson.customview.http.manager

import dinson.customview.kotlin.then
import dinson.customview.kotlin.times
import dinson.customview.kotlin.verbose
import dinson.customview.utils.NetworkUtils
import okhttp3.CacheControl
import okhttp3.Interceptor
import okhttp3.Response

/**
 *  日志拦截器
 */
class LoggingInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {

        verbose("┌${"─".times(120)}", showLine = false)
        verbose("│ ${chain.request()}", showLine = false)
        //verbose("│ ${request.headers().toString().split("\n".toRegex()).dropLastWhile { it.isEmpty() }}", showLine = false)
        val request = when {
            NetworkUtils.isNetworkAvailable() -> chain.request()
            else -> {
                verbose("│ Network error, then loading cache", showLine = false)
                chain.request().newBuilder().cacheControl(CacheControl.FORCE_CACHE).build()
            }
        }

        val t1 = System.nanoTime()
        val response = chain.proceed(request)
        val t2 = System.nanoTime()

        verbose("│ Received response for in ${(t2 - t1) / 1e6}ms", showLine = false)
        (response.headers().size() != 0) then {
            verbose("│ ${response.headers().toString().split("\n")}", showLine = false)
        }

        val mediaType = response.body()!!.contentType()
        val content = response.body()!!.string()
        if (content.isNotEmpty()) verbose("│ $content", showLine = false)
        verbose("└${"─".times(120)}", showLine = false)
        return response.newBuilder()
            .body(okhttp3.ResponseBody.create(mediaType, content))
            .build()
    }
}